sdk/avalon_sdk/worker/worker_details.py

```diff
@@ -205,9 +207,10 @@ def load_worker(self, worker_data):
             self.proof_data = json.loads(
                 worker_data['workerTypeData']['proofData'])
 
-        self.worker_id = crypto_utility.strip_begin_end_public_key(
+        w_id = crypto_utility.strip_begin_end_public_key(
             worker_data['workerTypeData']['verificationKey']) \
-            .encode("UTF-8").hex()
+            .encode("UTF-8")
+        self.worker_id = hashlib.sha256(w_id).hexdigest()
```

Comment:
Could you please add a comment what is this change for ? @rranjan3 had a similar done a change in a different PR.

Reply:
Done

Reply:
The worker_id stored in block chain as part of worker details should already be in 32 bytes as enclave manager is storing it in 32 bytes in shared KV DB -> https://github.com/hyperledger/avalon/blob/master/enclave_manager/avalon_enclave_manager/enclave_manager.py#L99. So why we need to duplicate the calculation of worker id again ?

Reply:
load_worker() function should take worker_id as argument instead of recalculating it here

Reply:
why do we need to pass worker_id to load_worker again? worker_data['workerTypeData']['verificationKey'] which is enclave public key, that we are converting it to sha256. Don't you think it is also duplicate? First of all why worker_id calculation is there in load_worker() earlier as well? Earlier also it was duplication calculation.

Reply:
Neater approach would be to use the worker_id from the worker_lookup() API rather than from the worker_details?

Reply:
Verification Key is the public part of signing key which is different from worker id.  So verification key is not duplicate. Earlier logic was also not correct. Can we fix that along with this bug fix PR ? Else please add a TODO comment in code and raise an issue which can be fixed in different PR.

Reply:
in load_worker(), we are calculating worker_id from ['workerTypeData']['verificationKey']. Why it was not taken care earlier itself? How come it is fine now if that logic is incorrect?

Reply:
Pass worker_id in constructor of workerDetails

Reply:
Created the issue to remote the worker id calculation in the sdk. https://github.com/hyperledger/avalon/issues/359

Reply:
> Pass worker_id in constructor of workerDetails

No need to worker_id in workerDetails. User should use the worker id got from worker_lookup().

Reply:
I mean when we crate WorkerDetails object, we should know for which worker_id this object is being created for. For that we can use worker_id we got from worker lookup.